functions/grades/workload.py

```diff
@@ -45,10 +45,17 @@ def app_handle(args, context, syscall):
         "push_date": context["push_date"]
     }
 
-    key = os.path.join(os.path.dirname(args["test_results"]),"grade.json")
-    syscall.write_key(bytes(key, "utf-8"), bytes(json.dumps(output), "utf-8"))
+    user = context['user']
+    func = context['function']
+    base_dir = os.path.join('/', func, user, context['metadata']['assignment'])
+    file = os.path.join(base_dir, 'grade.json')
+    syscall.endorse_with([[func]])
+    target_label = syscall.new_dclabel([[user]], [[func]])
```

Comment:
How does this `target_label` relate to the `current_label`? Ideally the current label in this function would more or have _at least_ the right secrecy component at this point (whatever data it reads should be exactly secret to the same people who can read the final grades)